spec/models/connected_app_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe ConnectedApp, type: :model do
+  it { is_expected.to belong_to :enterprise }
+
+  it "stores data as json hash" do
+    # This functionality is just Rails and would usually not warrant a spec but
+    # it's the first time we use the json datatype in this codebase and
+    # therefore it's a nice example to see how it works.
+    expect(subject.data).to eq nil
+
+    subject.enterprise = create(:enterprise)
+    subject.data = { link: "https://example.net" }
+    subject.save!
+    subject.reload
+
+    expect(subject.data).to eq({ "link" => "https://example.net" })
```

Comment:
Nice, I wonder if Rails has a way for it to be with_indifferent_access by default and therefore load the hash with symbol keys.

Reply:
We would need to add a few things:
https://gist.github.com/RobinDaugherty/5e7bc2a6e2369a29c77c

Or we just override a method on the model:
```rb
def data
  data.try(:with_indifferent_access)
end
```

Reply:
I looked at this and we currently have only one single line of access to this data:
```rb
enterprise.connected_apps[0].data&.fetch("link", false)
```

And in this scenario it would only change `"link"` to `:link` which isn't making a real difference. I'll leave this.